tuxemon/locale.py

```diff
@@ -247,29 +244,23 @@ def replace_text(session: Session, text: str) -> str:
 
     for i in range(len(player.monsters)):
         monster = player.monsters[i]
-        text = text.replace("${{monster_" + str(i) + "_name}}", monster.name)
-        text = text.replace(
-            "${{monster_" + str(i) + "_desc}}",
-            monster.description,
-        )
-        text = text.replace("${{monster_" + str(i) + "_type}}", monster.slug)
-        text = text.replace(
-            "${{monster_" + str(i) + "_category}}",
-            monster.category,
-        )
-        text = text.replace("${{monster_" + str(i) + "_shape}}", monster.shape)
-        text = text.replace(
-            "${{monster_" + str(i) + "_hp}}",
-            str(monster.current_hp),
-        )
-        text = text.replace(
-            "${{monster_" + str(i) + "_hp_max}}",
-            str(monster.hp),
-        )
-        text = text.replace(
-            "${{monster_" + str(i) + "_level}}",
-            str(monster.level),
-        )
+        _name = "${{monster_" + str(i) + "_name}}"
```

Comment:
I'm not sure this is an improvement. Saving the new strings to a var doesn't really help anything, since they're only used once. It's maybe *slightly* more aesthetic, but I think that's just a matter of opinion.

I'd like to get other people's opinion though. @vnmabus @bitcraft or @kerizane.

Reply:
I don't know, potentially we can remove everything in that loop (included the loop itself), because it's used only once in an unused map ("You watch some TV with your beloved Lv ${{monster_0_level}} ${{monster_0_name}}.,0")

Reply:
no, i mean that i don't think it makes sense to save the constructed strings to a local variable (_name, _desc, etc.) when they are used only once and on the very next line.

the previous implementation has the strings built in-line with the string replace

Reply:
ah ok, no problem

Reply:
I went back to the original structure, since I have an idea.